src/cli_plugin/install/progress.js

```diff
@@ -0,0 +1,39 @@
+/*
+Generates file transfer progress messages
+*/
+export default function Progress(logger) {
+  const self = this;
+
+  self.dotCount = 0;
+  self.runningTotal = 0;
+  self.totalSize = 0;
+  self.logger = logger;
```

Comment:
Rather than repeat this in each function, just consider that for the whole file.